magefile.go

```diff
@@ -152,7 +152,34 @@ func UpdateLibs() error {
 
 // E2e runs e2e tests with a built plugin. Requires docker-compose.
 func E2e() error {
-	return sh.RunV("docker-compose", "--file", "e2e/docker-compose.yml", "up", "--abort-on-container-exit")
+	envoyImagesList := strings.TrimSpace(os.Getenv("ENVOY_IMAGES"))
```

Comment:
I meant no change here at all. I guess we do want to avoid GH matrix build since the wasm filter build takes so long, but in that case let's just do something like this in the workflow, supporting multiple images makes this task much more complicated without much use IMO

```
for $image in $ENVOY_IMAGES; do
  ENVOY_IMAGE=$image go run mage.go e2e
done
```